File: ppmtool/src/main/java/com/lige110/springreact/ppmtool/repositories/ProjectRepository.java
package com.lige110.springreact.ppmtool.repositories;

import com.lige110.springreact.ppmtool.domain.Project;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ProjectRepository extends JpaRepository<Project, Long> {

    Project findByProjectIdentifier(String identifier);

    Iterable<Project> findAllByProjectLeader(String username);

}

File: src/main/kotlin/com/lige110/ppmtool/repositories/ProjectRepository.kt
package com.lige110.ppmtool.repositories

import com.lige110.ppmtool.domain.Project
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface ProjectRepository : CrudRepository<Project, Long> {
    fun findByProjectIdentifier(identifier: String): Project?
}
